src/components/grid/grid.pw.tsx

```diff
@@ -0,0 +1,533 @@
+  });
+
+  test("correctly renders grid items when each have responsive settings passed", async ({
+    mount,
+    page,
+  }) => {
+    await mount(
+      <GridContainer>
+        <GridItem
+          responsiveSettings={[
+            {
+              maxWidth: "1500px",
+              gridColumn: "1 / 7",
+              gridRow: "1 / 1",
+              alignSelf: "stretch",
+              justifySelf: "stretch",
+            },
+            {
+              maxWidth: "1300px",
+              gridColumn: "1 / 13",
+              gridRow: "1 / 1",
+              alignSelf: "stretch",
+              justifySelf: "stretch",
+            },
+            {
+              maxWidth: "900px",
+              gridColumn: "1 / 9",
+              gridRow: "2 / 2",
+              alignSelf: "stretch",
+              justifySelf: "stretch",
+            },
+          ]}
+        >
+          <Pod>Item 1</Pod>
+        </GridItem>
+        <GridItem
+          responsiveSettings={[
+            {
+              maxWidth: "1500px",
+              gridColumn: "6 / 13",
+              gridRow: "1 / 1",
+              alignSelf: "end",
+              justifySelf: "end",
+            },
+            {
+              maxWidth: "1300px",
+              gridColumn: "1 / 13",
+              gridRow: "2 / 2",
+              alignSelf: "end",
+              justifySelf: "end",
+            },
+            {
+              maxWidth: "900px",
+              gridColumn: "1 / 9",
+              gridRow: "3 / 3",
+              alignSelf: "end",
+              justifySelf: "end",
+            },
+          ]}
+        >
+          <Pod>Item 2</Pod>
+        </GridItem>
+        <GridItem
+          responsiveSettings={[
+            {
+              maxWidth: "1500px",
+              gridColumn: "1 / 13",
+              gridRow: "3 / 3",
+              alignSelf: "start",
+              justifySelf: "stretch",
+            },
+            {
+              maxWidth: "1300px",
+              gridColumn: "1 / 13",
+              gridRow: "2 / 2",
+              alignSelf: "start",
+              justifySelf: "stretch",
+            },
+            {
+              maxWidth: "900px",
+              gridColumn: "1 / 9",
+              gridRow: "3 / 3",
+              alignSelf: "start",
+              justifySelf: "stretch",
+            },
+          ]}
+        >
+          <Pod>Item 3</Pod>
+        </GridItem>
+      </GridContainer>
+    );
+    await page.setViewportSize({ width: 1400, height: 900 });
+
+    const firstGridItem = gridItem(page, 0);
+    const secondGridItem = gridItem(page, 1);
+    const thirdGridItem = gridItem(page, 2);
+
+    await expect(firstGridItem).toHaveCSS("grid-column", "1 / 7");
+    await expect(firstGridItem).toHaveCSS("grid-row", "1 / 1");
+    await expect(firstGridItem).toHaveCSS("align-self", "stretch");
+    await expect(firstGridItem).toHaveCSS("justify-self", "stretch");
```

Comment:
Should there be a check on the `width` value here? It's checked in the Cypress tests.

Reply:
Oh dear, you're right. Good spot 😄